src/gen_rpc_client.erl

```diff
@@ -581,3 +591,29 @@ maybe_start_client(NodeOrTuple) ->
         Pid ->
             {ok, Pid}
     end.
+
+%% Bypass the pipeline for local calls.
+%%
+%% Note: this function doesn't support authorization checks and/or
+%% module version checks.
+%%
+%% Note: `call_middleman' returns value by throwing an error... So we
+%% have to suppress dialyzer warning:
+-dialyzer({no_return, [local_call/4]}).
+local_call(M, F, A, undefined) ->
+    local_call(M, F, A, infinity);
+local_call({M, _Version}, F, A, Timeout) ->
+    local_call(M, F, A, Timeout);
+local_call(M, F, A, Timeout) ->
+    {Pid, MRef} = spawn_monitor(fun() ->
+                                        gen_rpc_acceptor:call_middleman(M, F, A)
+                                end),
+    receive
+        {'DOWN', MRef, process, Pid, {call_middleman_result, Reason}} ->
+            Reason;
+        {'DOWN', MRef, process, Pid, Other} ->
+            {badrpc, Other}
+    after Timeout ->
+            erlang:demonitor(MRef, [flush]),
+            {badrpc, timeout}
```

Comment:
Q: should we do one last check for the race condition that the response arrives between entering this block and de-monitoring?

Reply:
I think `flush` flag to demonitor takes care of this.